### DISCUSSION TITLE
How to replace existing string properly?

### DISCUSSION THREAD

**Discussion by phucly01** [olikraus/u8g2#2314] [Q&A]
I am using U8g2 and I can't seem to find a way to replace one character in a string.

For example, I do:

display.drawStr(10, 0, "Testing");
display.sendBuffer();

then later do:

display.drawStr(10, 0, "Test it");
display.sendBuffer();

The "Test it" is written on top of the "Testing" instead of clear the Testing then print Test it. 

Is there a way to clear the "area" with background color and then draw new string?

I tried print empty string but it didn't clear the area and I can't find any drawFilledRect() to do so.

**Comment by olikraus** [comment]
> Is there a way to clear the "area" with background color and then draw new string?

option 1: use font mode 0 (https://github.com/olikraus/u8g2/wiki/u8g2reference#setfontmode)
option 2: use drawBox() with color 0

**Reply by phucly01** [reply]
Thanks olikraus.  
I use the firstPage() nextPage() loop and it seems to be ok.  I was afraid the while loop holds the entire process but it isn't.  I will try your way sometime later.
